python/pyarrow/_dataset.pyx

```diff
@@ -3009,6 +3009,52 @@ def _get_partition_keys(Expression partition_expression):
     return out
 
 
+ctypedef CParquetFileWriter* _CParquetFileWriterPtr
+
+cdef class WrittenFile(_Weakrefable):
```

Comment:
Since this is essentially a named tuple with optional properties based on the format, I think it'd be better to just use a `dict` instead

Reply:
Hmm...I'm not sure I agree.  The same could be said of ReadOptions, ParseOptions, etc.  Since this is part of the public API and the available keys are well defined I think it best to create a class so it can be documented and communicated exactly what should be provided.